docs/content/latest/api/ysql/exprs/func_yb_hash_code.md

```diff
@@ -0,0 +1,195 @@
+---
+title: yb_hash_code() function [YSQL]
+headerTitle: yb_hash_code()
+linkTitle: yb_hash_code()
+description: Returns the partition hash code for a given set of expressions. 
+menu:
+  latest:
+    identifier: api-ysql-exprs-yb_hash_code
+    parent: api-ysql-exprs
+aliases:
+  - /latest/api/ysql/exprs/func_yb_hash_code
+isTocNested: true
+showAsideToc: true
+---
+
+## Synopsis
+`yb_hash_code` is a function that returns the hash of a set of given input values using the hash function DocDB uses to shard its data. In effect, it provides direct access to the hash value of any given row of a YSQL table, allowing one to infer a row’s physical location. This enables an application to specify queries based on the physical location of a row or set of rows. A user’s newfound access to the physical location of a row should open up possibilities for more efficient queries.
+
+## Instructions
+`yb_hash_code` can be used anywhere a normal function can be used in a YSQL expression. There is no limitation to where it can be used. It takes in a variable number of parameters whose types must be allowed in YB primary keys. More formally,
+
+```
+yb_hash_code(a1:t1, a2:t2, a3:t3, a4:t4…..) → int4 (32 bit integer)
+```
+
+Where `a1, a2, a3…` are expressions of type `t1, t2, t3…`, respectively. `t1, t2, t3…` must be types that are currently allowed in a primary key. 
+
+## Function Pushdown
+
+
+This function can be either evaluated at the Postgres layer after resolving each of its argument expressions, or certain invocations of it may be pushed down to be evaluated at the DocDB layer. We discuss the situations where they are pushed down. `yb_hash_code` invocations are pushed down when the Postgres optimizer determines that the return values of the function will directly match the values of the hash value column in a requested base table or index table. For example, if we create a table as follows
+
+```
+CREATE TABLE sample_table (x INTEGER, y INTEGER, z INTEGER, PRIMARY KEY((x,y) HASH), z ASC);
+```
+
+Then the following query will evaluate the `yb_hash_code` calls at the DocDB layer:
+
+```
+SELECT * FROM sample_table WHERE yb_hash_code(x,y) <= 128 AND yb_hash_code(x,y) >= 0;
+```
+
+You can expect to verify this with the `EXPLAIN ANALYZE` result of this statement
+```
+                                                            QUERY PLAN                                                             
+-----------------------------------------------------------------------------------------------------------------------------------
+ Index Scan using sample_table_pkey on sample_table  (cost=0.00..5.88 rows=16 width=12) (actual time=2.867..2.919 rows=18 loops=1)
+   Index Cond: ((yb_hash_code(x, y) <= 128) AND (yb_hash_code(x, y) >= 0))
+ Planning Time: 0.052 ms
+ Execution Time: 2.953 ms
+(4 rows)
+```
+
+Here, we see that the primary key index was used and no row was rechecked at the postgres layer.
```

Comment:
PostgreSQL is a trademark, so we should use it rather than nicknames. ("Postgres" with a capital P is also trademarked, but for consistency, I'm trying to get us using the full name.)
```suggestion
Here, we see that the primary key index was used and no row was rechecked at the PostgreSQL layer.
```